pkg/cloudprovider/aws/builder.go

```diff
@@ -30,7 +30,7 @@ func NewCloudProvider(logger logr.Logger) (cloudprovider.CloudProvider, error) {
 
 	var log = defaultLogger
 	if logger != nil {
```

Comment:
Possible to remove this logic? Make it so that you must provide a non-pointer logger to `NewCloudProvider`? That way we don't need the default logger logic. Should simplify things

Reply:
Will also remove all the places you've had to add `&logger`

Reply:
Yeah that works too, just wasn't sure if this removal of this logic was desired.
Much simpler not having it around